### PR TITLE
Refactor conversion of pairs of offsets

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
This way sharing code between new and old implementation should be easier.

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/index/LedgerBackedIndexService.scala [hunk 1]
```diff
@@ -124,47 +124,52 @@ abstract class LedgerBackedIndexService(
...
-            ledger
-              .ledgerEntries(Some(begin), endOpt)
-              .collect {
-                case (offset, t: LedgerEntry.Transaction) =>
-                  (toAbsolute(offset), t)
-              }
+            Source.failed(
+              ErrorFactories.invalidArgument(
+                s"End offset ${end.toApiString} is before Begin offset ${begin.toApiString}."))
+          case endOpt: Option[Offset] =>
```
Comment (S11001001) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/index/LedgerBackedIndexService.scala]: `:` doesn't start an ascription here, only a pattern.

```suggestion
          case endOpt =>
```

Comment (stefanobaghino-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/index/LedgerBackedIndexService.scala]: Sorry, but I don't know what 'ascription' means. Can you explain?

Comment (S11001001) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/index/LedgerBackedIndexService.scala]: Ascriptions (exp: t) are static; this syntax implies a dynamic, runtime check.

Comment (stefanobaghino-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/index/LedgerBackedIndexService.scala]: I see. I simply preserved the original, which was `None | Some(_)` and that was motivated [here](https://github.com/digital-asset/daml/pull/4949#discussion_r391514099). Thoughts?

**Pull request merged**